# Segmenting and clustering neighborhoods in Toronto, Canada : part 2/3

The cell below reuses the code from part 1 of this assignment to get toronto dataframe ready to use.

In [1]:
import numpy as np
import pandas as pd
url = 'https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M'
toronto_raw = pd.read_html(url)
toronto = toronto_raw[0]
toronto = toronto[toronto['Borough'] != "Not assigned"]
toronto.reset_index(drop=True, inplace=True)

In [2]:
# !pip install geocoder

In [3]:
import requests, geocoder

Note : as Google API did not return any result, I used arcgis instead.

In [4]:
lat_list = []
long_list = []
i = 1
for code in toronto['Postal Code']:
    g = geocoder.arcgis(f'{code}, Toronto, Ontario')
    lat_list.append(g.json['lat'])
    long_list.append(g.json['lng'])
    if i%10 == 0: print(f'{i}...', end = ' ', flush=True)
    i += 1
print('Done!')    

10... 20... 30... 40... 50... 60... 70... 80... 90... 100... Done!


In [ ]:
toronto.loc[:,'Latitude'] = lat_list
toronto.loc[:,'Longitude'] = long_list

In [6]:
toronto

Postal Code           Borough  \
0           M3A        North York   
1           M4A        North York   
2           M5A  Downtown Toronto   
3           M6A        North York   
4           M7A  Downtown Toronto   
..          ...               ...   
98          M8X         Etobicoke   
99          M4Y  Downtown Toronto   
100         M7Y      East Toronto   
101         M8Y         Etobicoke   
102         M8Z         Etobicoke   

                                         Neighbourhood  Latitude  Longitude  
0                                            Parkwoods  43.75245  -79.32991  
1                                     Victoria Village  43.73057  -79.31306  
2                            Regent Park, Harbourfront  43.65512  -79.36264  
3                     Lawrence Manor, Lawrence Heights  43.72327  -79.45042  
4          Queen's Park, Ontario Provincial Government  43.66253  -79.39188  
..                                                 ...       ...        ...  
98       The Kingsway, Montgomery Road, Old Mill North  43.65319  -79.51113  
99                                Church and Wellesley  43.66659  -79.38133  
100  Business reply mail Processing Centre, South C...  43.64869  -79.38544  
101  Old Mill South, King's Mill Park, Sunnylea, Hu...  43.63278  -79.48945  
102  Mimico NW, The Queensway West, South of Bloor,...  43.62513  -79.52681  

[103 rows x 5 columns]

Coordinates seem to be slightly different... (e.g : M5A Downton Toronto is 43.65512, -79.36264 instead of 43.654250, -79.360636 which represent about 190m).
Maybe because I did not use the Google API ?
To avoid any problem I would rather use the csv file data but it seems not to be available (Error 501).